index.js

```diff
@@ -25,6 +25,31 @@ module.exports = function (promises) {
     .then(function (res) {
       resolve(results);
     })
+    /*
+      The catch below makes sequential halt the execution of promises as soon as the first promise
+      is rejected or an error is thrown.
+
+      Previously, the code used to handle errors the promise sequence within the reduce function,
+      without re-throwing them. This hid the error from the 'reduce' function, which would continue
+      to execute promises as if nothing happened.
+
+      // underneath the hood the code looked like this
```

Comment:
Simplify this to get rid of the reduce and show the final promise chain. Given a promise sequence of `[ () => p1, ... ]`, we would generate `p1.catch().p2.catch().p3.catch()` which is why we would continue to execute.